gateway/join.go

```diff
@@ -133,17 +134,17 @@ func (g *Gateway) parseJoinRequestPacket(payload []byte) (joinRequest, *Device,
 // | MHDR | JOIN NONCE | NETID |   DEV ADDR  | DL | RX DELAY |   CFLIST   | MIC  |
 // | 1 B  |     3 B    |   3 B |     4 B     | 1B |    1B    |  0 or 16   | 4 B  |
 // https://lora-alliance.org/wp-content/uploads/2020/11/lorawan1.0.3.pdf page 35 for more info on join accept.
-func (d *Device) generateJoinAccept(ctx context.Context, jr joinRequest) ([]byte, error) {
+func generateJoinAccept(ctx context.Context, jr joinRequest, d *node.Node) ([]byte, error) {
 	// generate random join nonce.
 	jn := generateJoinNonce()
 
 	// generate a random device address to identify uplinks.
-	d.addr = generateDevAddr()
+	d.Addr = generateDevAddr()
```

Comment:
Would these need mutex protection when a node is reconfiguring? Do we anticipate an address mismatch ending up in a bad state where a node is on the old address but the gateway has rebuilt itself?

Reply:
thanks for pointing this out, I added reconfigure to gateway so it doesn't lose the device map if it rebuilds. Same with reconfiguring the node so that the dev addr can be saved from the gateway list if the node is reconfigured. This way the join process does not need to be redone if the node is reconfigured.

Reply:
Finally, a use-case for reconfigure that I don't hate!

 For whatever reason, we need more control over the gateway keeping track of the gateways nodes across multiple shared devices from the same process running a module -; we can keep track of them in a global variable, but I don't think that is something we want to ever try to do anyway .